basculeLogging.go

```diff
@@ -22,11 +27,33 @@ func sanitizeHeaders(headers http.Header) (filtered http.Header) {
 	return
 }
 
-func setLogger(logger log.Logger) func(delegate http.Handler) http.Handler {
+// func setLogger(logger log.Logger) func(delegate http.Handler) http.Handler {
+// 	return func(delegate http.Handler) http.Handler {
+// 		return http.HandlerFunc(
+// 			func(w http.ResponseWriter, r *http.Request) {
+// 				kvs := []interface{}{"requestHeaders", sanitizeHeaders(r.Header), "requestURL", r.URL.EscapedPath(), "method", r.Method}
+// 				kvs, _ = candlelight.AppendTraceInfo(r.Context(), kvs)
+// 				ctx := r.WithContext(logging.WithLogger(r.Context(), log.With(logger, kvs...)))
+// 				delegate.ServeHTTP(w, ctx)
+// 			})
+// 	}
+// }
+
+func setLogger(logger log.Logger, lf ...LoggerFunc) func(delegate http.Handler) http.Handler {
+
+	if logger == nil {
+		panic("The base Logger cannot be nil")
+	}
+
 	return func(delegate http.Handler) http.Handler {
 		return http.HandlerFunc(
 			func(w http.ResponseWriter, r *http.Request) {
 				kvs := []interface{}{"requestHeaders", sanitizeHeaders(r.Header), "requestURL", r.URL.EscapedPath(), "method", r.Method}
+				if len(lf) > 0 {
+					for _, f := range lf {
```

Comment:
We don't need an `if` check here because if `lf` has a length of 0, the for loop will be skipped over.

One thing I would add is a `nil` check for f...something like this:
```suggestion
				for _, f := range lf {
				    if f != nil {
				    // etc...
```